infrastructure/kubeflow/main.go

```diff
@@ -0,0 +1,219 @@
+				WaitForJobs: pulumi.Bool(true),
+				Values: pulumi.Map{
+					"global": pulumi.Map{
+						"proxy": pulumi.Map{
+							"autoInject": pulumi.String("disabled"),
+						},
+					},
+				},
+			}, pulumi.Provider(k8sProvider),
+				pulumi.DependsOn([]pulumi.Resource{istioBase}),
+				pulumi.DeleteBeforeReplace(true),
+				pulumi.RetainOnDelete(true))
+
+			if err == nil {
+				break
+			}
+
+			if i < maxRetries-1 {
+				ctx.Log.Warn(fmt.Sprintf("Istiod installation failed, retrying in 30 seconds (attempt %d of %d)", i+1, maxRetries), nil)
+				time.Sleep(30 * time.Second)
+			} else {
+				return fmt.Errorf("failed to install Istiod after %d attempts: %v", maxRetries, err)
+			}
+		}
+
+		// Install Cert-Manager
+		_, err = helm.NewRelease(ctx, "cert-manager", &helm.ReleaseArgs{
+			Chart:     pulumi.String("cert-manager"),
+			Version:   pulumi.String("v1.13.1"),
+			Namespace: pulumi.String("cert-manager"),
+			RepositoryOpts: helm.RepositoryOptsArgs{
+				Repo: pulumi.String("https://charts.jetstack.io"),
+			},
+			Values: pulumi.Map{
+				"installCRDs": pulumi.Bool(true),
+			},
+		}, pulumi.Provider(k8sProvider))
+		if err != nil {
+			return err
+		}
+
+		// Install Kyverno
+		_, err = helm.NewRelease(ctx, "kyverno", &helm.ReleaseArgs{
+			Chart:     pulumi.String("kyverno"),
+			Version:   pulumi.String("3.2.6"),
+			Namespace: pulumi.String("kyverno"),
+			RepositoryOpts: helm.RepositoryOptsArgs{
+				Repo: pulumi.String("https://kyverno.github.io/kyverno/"),
+			},
+		}, pulumi.Provider(k8sProvider))
+		if err != nil {
+			return err
+		}
+
+		_, err = helm.NewRelease(ctx, "argocd", &helm.ReleaseArgs{
+			Chart:     pulumi.String("argo-cd"),
+			Version:   pulumi.String("7.3.10"),
+			Namespace: pulumi.String("argocd"),
+			RepositoryOpts: helm.RepositoryOptsArgs{
+				Repo: pulumi.String("https://argoproj.github.io/argo-helm"),
+			},
+			Values: pulumi.Map{
+				"server": pulumi.Map{
+					"extraArgs": pulumi.Array{
+						// TODO we probalby want TLS
+						pulumi.String("--insecure"),
+					},
+				},
+				"configs": pulumi.Map{
+					"secret": pulumi.Map{
+						"argocdServerAdminPassword": argocdAdminPassword,
+					},
+				},
+			},
+		}, pulumi.Provider(k8sProvider))
+		if err != nil {
+			return err
+		}
+
+		// Create RDS instance for MySQL
+		db, err := rds.NewInstance(ctx, "kubeflow-mysql", &rds.InstanceArgs{
+			Engine:             pulumi.String("mysql"),
+			EngineVersion:      pulumi.String("8.0"),
+			InstanceClass:      pulumi.String("db.t3.micro"),
+			AllocatedStorage:   pulumi.Int(20),
+			Name:               pulumi.String("kubeflow"),
+			Username:           dbUsername,
+			Password:           dbPassword,
+			SkipFinalSnapshot:  pulumi.Bool(true),
+			PubliclyAccessible: pulumi.Bool(false),
+		})
+		if err != nil {
+			return err
+		}
+
+		// Create S3 bucket for object storage
+		bucket, err := s3.NewBucket(ctx, "kubeflow-storage", &s3.BucketArgs{
+			Bucket: pulumi.String("psf-kubeflow-artifacts"),
```

Comment:
This is fine but something we may want to think about is whether we want certain buckets to be "permanent" and live outside infrastructure creation/deletion.

Reply:
yea, let's do that for more fundamental buckets